server/src/main/java/org/candlepin/dto/api/v1/AsyncJobStatusTranslator.java

```diff
@@ -72,14 +71,54 @@ public AsyncJobStatusDTO populate(ModelTranslator translator, AsyncJobStatus sou
         destination.setAttempts(source.getAttempts());
         destination.setMaxAttempts(source.getMaxAttempts());
         destination.setResult(source.getJobResult());
+        destination.setState(stateToString(mapState(source.getState())));
+        destination.setPreviousState(stateToString(mapState(source.getPreviousState())));
 
-        JobState state = source.getState();
-        destination.setState(state != null ? state.name() : null);
+        return destination;
+    }
 
-        JobState pstate = source.getPreviousState();
-        destination.setPreviousState(pstate != null ? pstate.name() : null);
+    private String stateToString(PublicJobState state) {
+        if (state == null) {
+            return null;
+        }
+        return state.name();
+    }
 
-        return destination;
+    public static PublicJobState mapState(JobState state) {
+        if (state == null) {
+            return null;
+        }
+        PublicJobState publicJobState;
+        switch (state) {
+            case FINISHED:
+                publicJobState = PublicJobState.FINISHED;
+                break;
+            case CREATED:
+                publicJobState = PublicJobState.CREATED;
+                break;
+            case QUEUED:
+            case RUNNING:
+            case SCHEDULED:
+            case WAITING:
+                publicJobState = PublicJobState.RUNNING;
+                break;
+            case ABORTED:
+            case CANCELED:
+            case FAILED:
+            case FAILED_WITH_RETRY:
```

Comment:
I'm torn on this one. I can see a temporary "FAILED" output being useful, but it's still kind of "RUNNING" since it'll get picked up again unless something breaks internally.

Reply:
I haven't thought of that. It makes sense to map it to failed only once it reaches terminal state. Done.